### PR TITLE
Accounts module and logic bug fixes - Subtask #677 

### PR DESCRIPTION
**Pull request by SargeKhan**
Resolves issues mentioned in #677 except for `objectNormalize` issue, which will be fixed with @4miners accounts module rewrite PR

### CODE REVIEW

File: helpers/z_schema.js [hunk 1]
```diff
@@ -49,27 +49,15 @@ z_schema.registerFormat('username', function (str) {
...
 
 z_schema.registerFormat('hex', function (str) {
-	try {
-		Buffer.from(str, 'hex');
-	} catch (e) {
-		return false;
-	}
-
-	return true;
+	return /^[a-fA-F0-9]*$/.test(str);
```
Comment (MaciejBaj) [inline helpers/z_schema.js]: '' passes

Comment (MaciejBaj) [inline helpers/z_schema.js]: also, we are not checking length anywhere, what do you think about checking length if it fits in 32 or 64? I think we are not testing any different scenarios.

File: logic/account.js [hunk 2]
```diff
@@ -496,13 +496,11 @@ Account.prototype.verifyPublicKey = function (publicKey) {
...
-		if (publicKey.length < 64) {
+		if (publicKey.length !== 64) {
 			throw 'Invalid public key, must be 64 characters long';
 		}
-		// Check format
-		try {
-			Buffer.from(publicKey, 'hex');
-		} catch (e) {
+
+		if (!/^[a-fA-F0-9]+$/.test(publicKey)) {
```
Comment (MaciejBaj) [inline logic/account.js]: can we keep these checks in one place across the project? Same logic is used in z_schema file

File: helpers/z_schema.js (hunk 1 above)
Comment (SargeKhan) [inline helpers/z_schema.js]: This validator is only for hex strings. We have different validators for `publicKey` and `signature` which check whether the length is equal to 64 and 128.
An empty string was valid previously, so I didn't change it.

File: logic/account.js (hunk 2 above)
Comment (SargeKhan) [inline logic/account.js]: Yes, this seems like a good idea.

**Comment by SargeKhan** [comment]
@Isabello, @4miners I have removed [Account.internal.count](https://github.com/LiskHQ/lisk/pull/679/files#diff-08f191403f0b00d44cb59887bb15e8b5L546) and [Account.internal.getAllAccounts](https://github.com/LiskHQ/lisk/pull/679/files#diff-08f191403f0b00d44cb59887bb15e8b5L574) which didn't seem to return any useful information. How should we proceed considering @MaciejBaj's comment?

**Comment by Isabello** [comment]
getAll and count never worked correctly. Count might be a very heavy operation to serve up via API. I think removing these two things (which didn't work before) is fine. We can create new functionality for these later if necessary.

File: logic/account.js [hunk 3]
```diff
@@ -496,13 +496,11 @@ Account.prototype.verifyPublicKey = function (publicKey) {
 			throw 'Invalid public key, must be a string';
 		}
 		// Check length
-		if (publicKey.length < 64) {
```
Comment (4miners) [inline logic/account.js]: Why not replace that two checks with one `{format: 'publicKey'}`?

Comment (SargeKhan) [inline logic/account.js]: Didn't want to change the error thrown on these checks.

Comment (4miners) [inline logic/account.js]: ok

**Pull request merged**